src/py/flwr/server/compat/driver_client_proxy.py

```diff
@@ -168,8 +168,27 @@ def _send_receive_recordset(
             )
             if len(task_res_list) == 1:
                 task_res = task_res_list[0]
+                validate_task_res(task_res=task_res)
                 return serde.recordset_from_proto(task_res.task.recordset)
 
             if timeout is not None and time.time() > start_time + timeout:
                 raise RuntimeError("Timeout reached")
             time.sleep(SLEEP_TIME)
+
+
+def validate_task_res(
+    task_res: task_pb2.TaskRes,  # pylint: disable=E1101
+) -> None:
+    """Validate if a TaskRes is empty or not."""
+    if not task_res.HasField("task"):
+        raise ValueError("Invalid TaskRes, field `task` missing")
+    if not task_res.task.HasField("recordset"):
+        raise ValueError("Invalid Task, field `recordset` missing")
```

Comment:
A Error message will raise this since the content (i.e. `task.recordset` is empty)

Reply:
@jafermarq i pushed an update